### ISSUE TITLE
[BUG] Running a pipes flow on TaskServer throws error

### ISSUE THREAD

**Issue by grtjn** [marklogic-community/pipes#293]
**Describe the bug**

The following error is thrown when triggering the processing of a single file, and executing the DHF flow with multiple pipes steps on the TaskServer:

```
Error running step: XDMP-ARGTYPE: xdmp.type(null) -- arg1 is not of type xs.anyAtomicType. XDMP-ARGTYPE: xdmp.type(null) -- arg1 is not of type xs.anyAtomicType
in /custom-modules/pipes/core.sjs, at 776:13, in fn_head.onExecute() [javascript]
in /custom-modules/pipes/litegraph.sjs, at 781:19, in LGraph.runStep() [javascript]
in /custom-modules/pipes/litegraph.sjs, at 701:11, in LGraph.start() [javascript]
in /custom-modules/pipes/graphHelper.sjs, at 368:8, in executeGraphFromJson() [javascript]
in /custom-modules/pipes/graphHelper.sjs, at 314:9, in executeGraphStep() [javascript]
in /custom-modules/custom/pipesEntityEnrich/main.sjs, at 69:24, in main() [javascript]
in /data-hub/5/impl/flow.sjs, at 485:11, in runMain() [javascript]
in /data-hub/5/impl/flow.sjs, at 343:57, in runStep() [javascript]
in /data-hub/5/impl/flow.sjs, at 213:11, in runFlow() [javascript]
in /lib/flow-runner/step.sjs, at 24:24, in runStep() [javascript]
in /lib/flow-runner/flow.sjs, at 23:37, in runFlow() [javascript]
in /lib/clueyLib.sjs, at 98:19, in harmonizeClueyObservation() [javascript]
in /triggers/cluey-trigger.sjs, at 13:13 [javascript]
in /data-hub/5/impl/invoke-step.sjs,
in xdmp:invoke(\"/data-hub/5/impl/invoke-step.sjs\", {flow:{config:{STAGINGDATABASE:\"data-hub-STAGING\", FINALDATABASE:\"data-hub-FINAL\", TRACEDATABASE:\"data-hub-JOBS\", ...}, datahub:{config:{STAGINGDATABASE:\"data-hub-STAGING\", FINALDATABASE:\"data-hub-FINAL\", TRACEDATABASE:\"data-hub-JOBS\", ...}, consts:{XQUERY:\"xqy\", JAVASCRIPT:\"sjs\", XML:\"xml\", ...}, debug:{logLevel:\"default\", logLevels:[\"default\", \"notice\", \"trace\", ...], logTypes:[\"error\", \"warning\", \"notice\", ...]}, ...}, step:{config:{STAGINGDATABASE:\"data-hub-STAGING\", FINALDATABASE:\"data-hub-FINAL\", TRACEDATABASE:\"data-hub-JOBS\", ...}, hubUtils:{config:{STAGINGDATABASE:\"data-hub-STAGING\", FINALDATABASE:\"data-hub-FINAL\", TRACEDATABASE:\"data-hub-JOBS\", ...}}, performance:{config:{}, dataHubMetrics:[], stepMetrics:[]}, ...}, ...}, uris:[\"/Observation/3718716/infrastructure/1d627b16-15bf-4aba-b0e2-10a0...\"], content:[{uri:\"/Observation/3718716/infrastructure/1d627b16-15bf-4aba-b0e2-10a0...\", value:cts.doc(\"/Observation/3718716/infrastructure/1d627b16-15bf-4aba-b0e2-10a090f6df33/2139700884107940997.json\"), context:{permissions:[{capability:\"update\", roleId:\"9159348592924085917\"}, {capability:\"read\", roleId:\"9159348592924085917\"}, {capability:\"read\", roleId:\"1126085831936416318\"}, ...], metadata:{datahubCreatedBy:\"geert\", datahubCreatedByStep:\"pipesClueyToObservationEntity\", datahubCreatedInFlow:\"process-ClueyObservations\", ...}, originalCollections:[\"pipesClueyToObservationEntity\", \"sccss-cluey\", \"Observation\", ...]}}], ...}, {database:\"5221970715559444025\", update:\"false\", commit:\"auto\", ...}) [javascript]
in /triggers/cluey-trigger.sjs [javascript]
in /triggers/cluey-trigger.sjs [javascript]
```

**To Reproduce**

Difficult to isolate. When running the same pipes steps from pipes ui, everything runs fine. Running flow from gradle (effectively on E-node, rather than on TaskServer), everything runs fine too.

**Expected behavior**

No difference however pipes is run.

**Screenshots**

n/a

**Desktop (please complete the following information):**

not relevant

**Additional context**

Server is running MarkLogic 10.0-4 CentOS Docker image.

**Comment by grtjn** [comment]
I suspect there is a very subtle bug involved in the integration of the Google JS Engine into MarkLogic core, but I found a way to bypass the issue, by using different code in litegraph.js

**Comment by grtjn** [comment]
The ultimate culprit was that `new Function(code)` was returning `null` instead of a function object, while there are no conditions imaginable in which that should happen. I'll open a PR that will eliminate the use of `Function`, which is recommended anyhow.

**Comment by josvanroosmalen** [comment]
Shipping this bug. Fixed in 1.4
